modules/library/src/main/scala/zio/elasticsearch/result/ElasticResult.scala

```diff
@@ -74,7 +74,7 @@ final class SearchResult private[elasticsearch] (
 
   lazy val response: UIO[SearchWithAggregationsResponse] = ZIO.succeed(fullResponse)
 
-  lazy val total: UIO[Long] = ZIO.succeed(fullResponse.hits.total.value)
+  lazy val total: IO[Option[Nothing], Long] = ZIO.fromOption(fullResponse.hits.total).map(_.value)
```

Comment:
I don't like this return type - Option[Nothing], is there any chance we can change this?